src/EditorFeatures/Core/Implementation/RenameTracking/RenameTrackingTaggerProvider.RenameTrackingCodeAction.cs

```diff
@@ -21,62 +22,82 @@ private class RenameTrackingCodeAction : CodeAction
             private readonly string _title;
             private readonly Document _document;
             private readonly IEnumerable<IRefactorNotifyService> _refactorNotifyServices;
-            private readonly bool _showPreview;
             private readonly ITextUndoHistoryRegistry _undoHistoryRegistry;
+            private RenameTrackingCommitter _renameTrackingCommitter;
 
-            public RenameTrackingCodeAction(Document document, string title, IEnumerable<IRefactorNotifyService> refactorNotifyServices, ITextUndoHistoryRegistry undoHistoryRegistry, bool showPreview)
+            public RenameTrackingCodeAction(Document document, string title, IEnumerable<IRefactorNotifyService> refactorNotifyServices, ITextUndoHistoryRegistry undoHistoryRegistry)
             {
                 _document = document;
                 _title = title;
                 _refactorNotifyServices = refactorNotifyServices;
                 _undoHistoryRegistry = undoHistoryRegistry;
-                _showPreview = showPreview;
             }
 
             public override string Title { get { return _title; } }
 
             protected override Task<IEnumerable<CodeActionOperation>> ComputeOperationsAsync(CancellationToken cancellationToken)
             {
+                // Invoked directly without previewing.
+                if (_renameTrackingCommitter == null)
+                {
+                    if (!TryInitializeRenameTrackingCommitter(cancellationToken))
+                    {
+                        return SpecializedTasks.EmptyEnumerable<CodeActionOperation>();
+                    }
+
+                    ////_renameTrackingCommitter.RenameSymbolAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+                }
+
+                var committerOperation = new RenameTrackingCommitterOperation(_renameTrackingCommitter);
+                return Task.FromResult(SpecializedCollections.SingletonEnumerable(committerOperation as CodeActionOperation));
+            }
+
+            protected override async Task<IEnumerable<CodeActionOperation>> ComputePreviewOperationsAsync(CancellationToken cancellationToken)
+            {
+                if (!_document.Project.Solution.Workspace.Options.GetOption(FeatureOnOffOptions.RenameTrackingPreview, _document.Project.Language) ||
```

Comment:
My understanding is that we need it. There was an ask to expose this option during the last "Preview" release timeframe and I think we intend to keep this. @Pilchie ?